spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ImportAutoConfigurationImportSelector.java

```diff
@@ -58,8 +58,7 @@
 
 	@Override
 	public Set<Object> determineImports(AnnotationMetadata metadata) {
-		Set<String> result = new LinkedHashSet<>();
-		result.addAll(getCandidateConfigurations(metadata, null));
+		Set<String> result = new LinkedHashSet<>(getCandidateConfigurations(metadata, null));
```

Comment:
I find personally the previous version more readable: nesting the method call in the constructor parameter makes it harder to read.

Reply:
Create the hashset with default size ,and then put all the item in it.Would not it cause the hash table resize?